project_euler/problem_206/sol1.py

```diff
@@ -0,0 +1,65 @@
+"""
+Project Euler 206
+https://projecteuler.net/problem=206
+
+Find the unique positive integer whose square has the form 1_2_3_4_5_6_7_8_9_0,
+where each “_” is a single digit.
+
+-----
+
+Instead of computing every single permutation of that number and going
+through a 10^9 search space, we can narrow it down considerably.
+
+If the square ends in a 0, then the square root must also end in a 0. Thus,
+the last missing digit must be 0 and the square root is a multiple of 10.
+We can narrow the search space down to the first 8 digits and multiply the
+result of that by 10 at the end.
+
+Now the last digit is a 9, which can only happen if the square root ends
+in a 3 or 7. We can either start checking for the square root from
+101010103, which is the closest square root of 10203040506070809 that ends
+in 3 or 7, or 138902663, the closest square root of 1929394959697989. The
+problem says there's only 1 answer, so starting at either point is fine,
+but the result happens to be much closer to the latter.
```

Comment:
Starting from `num = 101010103` gives me `470525830` which is incorrect. Also, without knowing the answer you cannot deduce whether to start from one number or the other. Please improve your solution.

Reply:
There's only a certain range of numbers that can fit the format, so you can make a few deductions from that. Starting at 101010103 means you have to increment `num` because it has the square closest to the lowest possible one fitting the format (10203040506070809), and starting at 138902663 means you have to decrement `num` because it has the highest possible square (1929394959697989). Here, the solution does the latter because from testing both options, the runtime is quite poor if you start at 101010103. You'll get the wrong answer if you try running this solution as is with `num < answer` because it's decrementing `num`, so you're moving away from the range the answer must be in.

I have an idea for introducing input, but it may drastically worsen the runtime. Is that ok?

Reply:
@dhruvmanila Do you have any preference for the two options in my last comment? Also, my other 2 open PRs #3035 and #3041 are ready for review.

Reply:
Yes, please mention this in the module-level docstring and also change the order of function so that `solution()` is at the end.